epitator/structured_incident_annotator.py

```diff
@@ -0,0 +1,287 @@
+            # for row in rows:
+            #     if len(row) > median_num_cols:
+            #         combine cols with unexpected types
+            #     elif len(row) < median_num_cols:
+            #         add empty cols or remove row
+            #     else:
+            #         good
+
+            # prior_table = None
+            # if len(tables) > 0:
+            #     prior_table = tables[-1]
+            # //combine with previous table
+            # if not has_header and prior_table and len(prior_table.header) == median_num_cols:
+            #     # TODO: Check matching column types
+
+            rows = zip(*parsed_column_entities)
+            tables.append(Table(
+                column_definitions,
+                rows,
+                metadata=dict(
+                    __section_title=__section_title,
+                    __last_geoname_mentioned=__last_geoname_mentioned,
+                    __last_date_mentioned=__last_date_mentioned)))
+
+        incidents = []
+        for table in tables:
+            for row_idx, row in enumerate(table.rows):
+                row_incident_date = table.metadata.get('__last_date_mentioned')
+                row_incident_location = table.metadata.get('__last_geoname_mentioned')
+                row_incident_base_type = None
+                row_incident_status = None
+                row_incident_aggregation = None
+                for column, value in zip(table.column_definitions, row):
+                    if not value: continue
+                    if column['type'] == 'date':
+                        row_incident_date = value
+                    elif column['type'] == 'geoname':
+                        row_incident_location = value
+                    elif column['type'] == 'incident_type':
+                        if "case" in value.text.lower():
+                            row_incident_base_type = "caseCount"
+                        elif "death" in value.text.lower():
+                            row_incident_base_type = "deathCount"
+                    elif column['type'] == 'incident_status':
+                        row_incident_status = value.text
+
+                row_incidents = []
+                for column, value in zip(table.column_definitions, row):
+                    if not value: continue
+                    if column['type'] == "number":
+                        column_name = column.get('name', '').lower()
+                        if row_incident_base_type:
+                            incident_base_type = row_incident_base_type
+                        else:
+                            if "cases" in column_name:
+                                incident_base_type = "caseCount"
+                            elif "deaths" in column_name:
+                                incident_base_type = "deathCount"
+                            else:
+                                incident_base_type = "caseCount"
```

Comment:
We're assuming here that all numeric columns are case or death counts? I know that other types of things in ProMED tables include culled animals.

Reply:
That's a good point. I'll add some code and tests to deal with those scenarios.

Reply:
It might be just reasonable to have an "other" to capture things we aren't explicitly looking for and avoiding populating junk values